src/mon/MonClient.cc

```diff
@@ -874,53 +889,45 @@ void MonClient::_finish_hunting(int auth_err)
   }
   _resend_mon_commands();
   send_log(true);
-  if (active_con) {
-    std::swap(auth, active_con->get_auth());
-    if (global_id && global_id != active_con->get_global_id()) {
-      lderr(cct) << __func__ << " global_id changed from " << global_id
-		 << " to " << active_con->get_global_id() << dendl;
-    }
-    global_id = active_con->get_global_id();
-  }
   if (sub.reload()) {
     _renew_subs();
   }
+  ceph_assert(active_con);
+  std::swap(auth, active_con->get_auth());
+  if (global_id && global_id != active_con->get_global_id()) {
+    lderr(cct) << __func__ << " global_id changed from " << global_id
+	       << " to " << active_con->get_global_id() << dendl;
+  }
+  global_id = active_con->get_global_id();
+  do_send_ping = make_lambda_context(
+    [this](int) {
+      send_ping();
+      _maybe_renew_subs();
+      _check_auth_tickets();
+    });
+  timer.add_event_after(cct->_conf->mon_client_ping_interval, do_send_ping);
+  do_send_log = make_lambda_context([this](int) { send_log(); });
+  timer.add_event_after(cct->_conf->mon_client_log_interval, do_send_log);
```

Comment:
I'm not seeing where another event is scheduled when do_send_ping or do_send_log runs?

Reply:
updated and repushed.